frontend/src/metabase/visualizations/visualizations/SmartScalar/SettingsComponents/PeriodsAgoOptionComponent.tsx

```diff
@@ -0,0 +1,158 @@
+import { useCallback, useMemo, useRef, useState } from "react";
+import type {
+  Dispatch,
+  KeyboardEvent,
+  MouseEvent,
+  SetStateAction,
+} from "react";
+import { Group, Text, Box } from "metabase/ui";
+import type { SelectedComparisonPeriodsAgo } from "metabase-types/api";
+import { NumberInputStyled } from "./PeriodsAgoOptionComponent.styled";
+import { MenuItemStyled } from "./MenuItem.styled";
+
+type PeriodsAgoOptionComponentProps = {
+  isSelected: boolean;
+  maxValue: number;
+  name: string;
+  onChange: (setting: { type: string; value?: number }) => void;
+  selectedValue?: SelectedComparisonPeriodsAgo;
+  setOpen: Dispatch<SetStateAction<boolean>>;
+  type: string;
+};
+
+export function PeriodsAgoOptionComponent({
```

Comment:
nit: I think it's clear it's a component and we don't need the word "Component" here and in the filename